packages/apidom-ns-json-schema-draft-4/src/refractor/visitors/json-schema/index.ts

```diff
@@ -1,33 +1,95 @@
 import { Mixin } from 'ts-mixer';
-import { always } from 'ramda';
+import { always, defaultTo } from 'ramda';
+import { isNonEmptyString, isUndefined } from 'ramda-adjunct';
+import {
+  ObjectElement,
+  ArrayElement,
+  isStringElement,
+  cloneDeep,
+  toValue,
+} from '@swagger-api/apidom-core';
 
 import FixedFieldsVisitor, {
   FixedFieldsVisitorOptions,
   SpecPath,
 } from '../generics/FixedFieldsVisitor.ts';
+import ParentSchemaAwareVisitor, {
+  ParentSchemaAwareVisitorOptions,
+} from './ParentSchemaAwareVisitor.ts';
 import FallbackVisitor, { FallbackVisitorOptions } from '../FallbackVisitor.ts';
 import JSONSchemaElement from '../../../elements/JSONSchema.ts';
+import { isJSONSchemaElement } from '../../../predicates.ts';
 
 /**
  * @public
  */
 export interface JSONSchemaVisitorOptions
   extends FixedFieldsVisitorOptions,
+    ParentSchemaAwareVisitorOptions,
     FallbackVisitorOptions {}
 
 /**
  * @public
  */
-class JSONSchemaVisitor extends Mixin(FixedFieldsVisitor, FallbackVisitor) {
+class JSONSchemaVisitor extends Mixin(
+  FixedFieldsVisitor,
+  ParentSchemaAwareVisitor,
+  FallbackVisitor,
+) {
   declare public element: JSONSchemaElement;
 
   declare protected readonly specPath: SpecPath<['document', 'objects', 'JSONSchema']>;
 
+  protected readonly default$schema = 'http://json-schema.org/draft-04/schema#';
```

Comment:
Let's redefine this as `get` accessory so that we have ability to compute a value, not just only statically assign it.